pkgs/servers/mail/sympa/default.nix

```diff
@@ -0,0 +1,120 @@
+{ stdenv, perlPackages, fetchFromGitHub, makeWrapper, autoreconfHook
+, dataDir ? "/var/lib/sympa"
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sympa";
+  version = "6.2.44";
+
+  src = fetchFromGitHub {
+    owner = "sympa-community";
+    repo = pname;
+    rev = version;
+    sha256 = "1vlv2cw5cz0xd080x1wpg8vxrr8bg1slvc6b59ciabnqivf3xpcw";
+  };
+
+  configureFlags = [
+    "--without-initdir"
+    "--without-unitsdir"
+    "--without-smrshdir"
+
+    "--with-lockdir=/run/sympa"
+    "--with-piddir=/run/sympa"
+    "--with-confdir=${dataDir}/etc"
+    "--sysconfdir=${dataDir}/etc"
+    "--with-spooldir=${dataDir}/spool"
+    "--with-expldir=${dataDir}/list_data"
+  ];
+  nativeBuildInputs = [ autoreconfHook ];
+  buildInputs = [ perlPackages.perl makeWrapper ];
+  propagatedBuildInputs = with perlPackages; [
```

Comment:
Wow! Are all of these direct dependencies of every program? This could probably benefit from using `perl.withPackages` like so:
```
prog1Env = perl.withPackages (p: with p; [ ArchiveZip CGI DateTime ... ]);
prog2Env = perl.withPackages (p: with p; [ DBI FGCI ]);
```

Note you can explicitly list dependencies for each program and a 'better' perl environment will be built. From there you can pass `prog1Env` and `prog2Env` into `buildInputs` and `patchShebangs` can take care of the `perl` binary.

Reply:
Don't know ... I'm not familiar with Sympa internals (nor can I read Perl very well) but it seems that the programs mostly depend on Sympa library/package which in turn depends on these packages.

Is there a way to determine which program depends on which package that doesn't involve reviewing the whole source code?

Reply:
Sorry I suppose I shouldn't have bothered mentioning about multiple `perl` environments.

The point you should takeaway is that you can avoid `PERL5LIB` if you want to by creating a dedicated (custom) `perl` environment using `perl.withPackages`, if you want to.

Reply:
I'm not sure I understand how that works. Do I define a custom `perlEnv` and then use it instead of `perlPackages.perl` in `buildInputs` and use `patchShebangs` instead of `makeWrapper`? How do I pass the environment to `patchShebangs`? Will `propagatedBuildInputs` be empty? Got an example I can look at?

Reply:
https://github.com/NixOS/nixpkgs/pull/64646 is a simple example which should help.

Reply:
Modified the package to use `perl.withPackages`, please take a look if this is what you meant.